### ISSUE TITLE
🚀 [Feature Request] Docker and docker-compose for cmangos-classic

### ISSUE THREAD

**Issue by s3v1** [cmangos/issues#2505]
## 🚀 Feature
I'd like to be able to build and run a cmangos-classic server using docker and docker-compose.

### Benefits & Purpose
To lower the "barrier of entry" and make it easy to run a cmangos-classic server. It would be so nice to be able to just clone the https://github.com/cmangos/mangos-classic repo and just run "docker-compose up", to get a running server.

The whole sequence of build, extract, make db, run does have a few moving parts. The easy part is making some docker/compose files that build and run the server, that hard part is getting the right data into the database and making the extracted files available.
I'd want to script and dockerize that  process too.

_I'm aware that the files extracted from the client cannot be added to the repo, due to copyright._

### Volunteering
I'd like to contribute this myself. I have good docker and linux skills from work, and I believe I can make something cool.
I would also add github actions to test the build out as part of every commit, just like you already do with the linux and windows builds.

**I just want to know in advance, if this is something you'd consider merging back into master. It would be sad for me to spend time and effort if you don't like the idea.**

Do you like the idea, or should I take another approach?

**Comment by berserkingyadis** [comment]
Lowering the barrier of entry is always good to make the cmangos more appealing to aspiring contributors. Docker sounds good for a headless project like this. I recall several moments where docker was being discussed in the discord and people being open towards it. One thing to consider is figuring out how to get the maps extracted and into the docker "system" since that is one external source and preferably part of the "one click solution".

**Comment by s3v1** [comment]
I'm afraid I have to close this feature request. I missed my window of opportunity during the summer vacation. Now it's back to work and I've run out of free time to do the work. Mayne I'll be back next holiday ;)

**Comment by deiteris** [comment]
But why don't keep it open so it is visible and maybe someone else can try implement this? 🤔

**Comment by Muehe** [comment]
@deiteris Good point, reopening this as a representative Docker issue, as this has been repeatedly asked for and tried out over the years.

@s3v1 No worries, this project is over a decade old, if you get around to it during next years vacation everybody will still be grateful. Or somebody else will pick it up.

As for the general issue at hand I'd second @berserkingyadis in saying search the Discord server. Most of the work has already been done and shared at one point or the other.

For the meta, we can't share extracted maps due to legal reasons, so pointing the Docker image to a client would necessarily be part of the solution. Also creating a new repo would possibly be prudent (at least at first) since it makes the effort independent from being merged and it's a way around dealing with the split in core/DB repo which you would have to deal with anyway (maybe by using a git submodule or similar).

**Comment by kderholt** [comment]
I have made a docker-compose version of all the 3 expansions. 
Its more or less "one command run"

https://github.com/kderholt/wowpserver

Try it out, feedback is appreciated

**Comment by cbrgm** [comment]
I also wrote a Dockerfile and docker-compose a while ago for cmangos-wotlk. Should still be working as it's always using the latest version from main branch. I deployed it on a k8s cluster back in the days and it worked all fine!

https://github.com/cbrgm/cmangos-docker

**Comment by Muehe** [comment]
Here is another Docker project that has gained some popularity recently:

https://github.com/korhaldragonir/cmangos-docker-classic
https://github.com/korhaldragonir/cmangos-docker-tbc
https://github.com/korhaldragonir/cmangos-docker (wotlk)

**Comment by Byloth** [comment]
Here's my project that is actually quite complete (for all `classic`, `tbc` and `wotlk` versions):
https://github.com/Byloth/cmangos-docker

It may need help for a better wiki and some minor stuff...
But I'm currently using without any major flaws! 🙃
